ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/TracingOptionsManager.java

```diff
@@ -36,69 +38,56 @@
-	private Properties template;
+	private Map<String, String> template;
 
 	public TracingOptionsManager() {
 		super();
 	}
 
-	private Properties createTemplate() {
-		Properties temp = new Properties();
-		IPluginModelBase[] models = PluginRegistry.getAllModels();
-		for (IPluginModelBase model : models) {
-			addToTemplate(temp, model);
-		}
-		return temp;
-	}
-
-	private void addToTemplate(Properties temp, IPluginModelBase model) {
-		Properties modelOptions = getOptions(model);
-		if (modelOptions != null) {
-			temp.putAll(modelOptions);
-		}
-	}
-
-	public Hashtable<String, String> getTemplateTable(String pluginId) {
-		Properties tracingTemplate = getTracingTemplate();
-		Hashtable<String, String> defaults = new Hashtable<>();
-		for (Enumeration<Object> keys = tracingTemplate.keys(); keys.hasMoreElements();) {
-			String key = keys.nextElement().toString();
+	public Map<String, String> getTemplateTable(String pluginId) {
+		Map<String, String> tracingTemplate = getTracingTemplate();
+		Map<String, String> defaults = new HashMap<>();
+		tracingTemplate.forEach((key, value) -> {
 			if (belongsTo(key, pluginId)) {
-				defaults.put(key, tracingTemplate.get(key).toString());
+				defaults.put(key, value);
 			}
-		}
+		});
 		return defaults;
 	}
 
 	private boolean belongsTo(String option, String pluginId) {
-		IPath path = IPath.fromOSString(option);
-		String firstSegment = path.segment(0);
+		String firstSegment = IPath.fromOSString(option).segment(0);
 		return pluginId.equalsIgnoreCase(firstSegment);
 	}
 
-	public Properties getTracingOptions(Map<String, String> storedOptions) {
+	public Map<String, String> getTracingOptions(Map<String, String> storedOptions) {
 		// Start with the fresh template from plugins
-		Properties defaults = getTracingTemplateCopy();
+		Map<String, String> defaults = getTracingTemplateCopy();
 		if (storedOptions != null) {
 			// Load stored values, but only for existing keys
-			Iterator<String> iter = storedOptions.keySet().iterator();
-			while (iter.hasNext()) {
-				String key = iter.next();
+			storedOptions.forEach((key, value) -> {
 				if (defaults.containsKey(key)) {
-					defaults.setProperty(key, storedOptions.get(key));
+					defaults.put(key, value);
 				}
-			}
+			});
 		}
 		return defaults;
 	}
 
-	public Properties getTracingTemplateCopy() {
-		return (Properties) getTracingTemplate().clone();
+	public Map<String, String> getTracingTemplateCopy() {
+		return new HashMap<>(getTracingTemplate());
 	}
 
-	private synchronized Properties getTracingTemplate() {
+	private synchronized Map<String, String> getTracingTemplate() {
 		if (template == null) {
-			template = createTemplate();
+			Map<String, String> temp = new HashMap<>();
+			IPluginModelBase[] models = PluginRegistry.getAllModels();
+			Arrays.stream(models).map(TracingOptionsManager::getOptions).filter(Objects::nonNull).forEach(p -> {
+				@SuppressWarnings({ "rawtypes", "unchecked" })
+				Map<String, String> entries = (Map) p;
```

Comment:
From this discussion I assume, that this is a save cast:
https://stackoverflow.com/q/873510/14542697

Reply:
Thanks for verifying this!